hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/WindowGroupAggregateTest.java

```diff
@@ -287,35 +287,40 @@ private void testSessionWindow(long earlyResultsPeriod) {
                 .map(ts -> ts + (ts / sessionLength) * sessionTimeout)
                 .collect(toList());
         addToSrcMapJournal(input);
-        if (earlyResultsPeriod == 0) {
+        boolean emittingEarlyResults = earlyResultsPeriod != 0;
+        if (!emittingEarlyResults) {
             addToSrcMapJournal(closingItems);
         }
 
         // When
-        Function<TimestampedEntry<String, Long>, String> formatFn = fx.tsEntryFormatFn;
-        StreamStage<TimestampedEntry<String, Long>> aggregated = fx.stage0()
-                .window(session(sessionTimeout)
-                        .setEarlyResultsPeriod(earlyResultsPeriod))
-                .groupingKey(Entry::getKey)
-                .aggregate(summingLong(Entry::getValue), (start, end, key, sum) ->
-                        // filter out one of the windows - this is to test map-to-null behavior
-                        start == 0 ? null : new TimestampedEntry<>(start, key, sum));
+        StageWithKeyAndWindow<Entry<String, Integer>, String> windowed = fx
+                .stage0()
+                .window(session(sessionTimeout).setEarlyResultsPeriod(earlyResultsPeriod))
+                .groupingKey(Entry::getKey);
 
         // Then
-        aggregated.drainTo(sink);
+
+        windowed.aggregate(summingLong(Entry::getValue), (start, end, key, sum) ->
+                (start == FILTERED_OUT_WINDOW_START || (emittingEarlyResults && end - start != sessionLength + 1))
```

Comment:
Why do you filter out results with just single event?

Reply:
I filter out all window results that are missing some events